akka-http-core/src/main/scala/akka/http/scaladsl/ClientTransport.scala

```diff
@@ -17,23 +17,28 @@ import scala.concurrent.Future
 
 /**
  * Abstraction to allow the creation of alternative transports to run HTTP on.
+ *
+ * (Still unstable) SPI for implementors of custom client transports.
  */
 @ApiMayChange
 trait ClientTransport { outer ⇒
-  def connectTo(host: String, port: Int)(implicit system: ActorSystem): Flow[ByteString, ByteString, Future[OutgoingConnection]]
+  def connectTo(host: String, port: Int, settings: ClientConnectionSettings)(implicit system: ActorSystem): Flow[ByteString, ByteString, Future[OutgoingConnection]]
 }
 
+/**
+ * (Still unstable) entry point to create or access predefined client transports.
+ */
+@ApiMayChange
 object ClientTransport {
-  def TCP(localAddress: Option[InetSocketAddress], settings: ClientConnectionSettings): ClientTransport =
-    new TCPTransport(localAddress, settings)
+  val TCP: ClientTransport = TCPTransport
 
-  private case class TCPTransport(localAddress: Option[InetSocketAddress], settings: ClientConnectionSettings) extends ClientTransport {
-    def connectTo(host: String, port: Int)(implicit system: ActorSystem): Flow[ByteString, ByteString, Future[OutgoingConnection]] =
+  private case object TCPTransport extends ClientTransport {
+    def connectTo(host: String, port: Int, settings: ClientConnectionSettings)(implicit system: ActorSystem): Flow[ByteString, ByteString, Future[OutgoingConnection]] =
```

Comment:
Yeah, this solves the problem of it being passed in "too early and only once", good.